akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -71,7 +72,8 @@ class HttpExt(config: Config)(implicit system: ActorSystem) extends akka.actor.E
     connections.map {
       case Tcp.IncomingConnection(localAddress, remoteAddress, flow) ⇒
         val layer = serverLayer(settings, Some(remoteAddress), log)
-        IncomingConnection(localAddress, remoteAddress, layer atop tlsStage join flow)
+        val flowWithTimeoutRecovered = flow.recover { case t: TimeoutException ⇒ throw new HttpConnectionTimeoutException(t.getMessage) }
```

Comment:
It still logs an error because the code throws out of the `recover` blog. What would be needed is a `mapFailure` combinator instead.